integration/spark/vendor/snowflake/src/main/java/io/openlineage/spark/agent/vendor/snowflake/SnowflakeVendor.java

```diff
@@ -0,0 +1,57 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.vendor.snowflake;
+
+import static io.openlineage.spark.agent.vendor.snowflake.Constants.SNOWFLAKE_PROVIDER_CLASS_NAME;
+
+import io.openlineage.spark.agent.lifecycle.VisitorFactory;
+import io.openlineage.spark.agent.vendor.snowflake.lifecycle.SnowflakeRelationVisitor;
+import io.openlineage.spark.agent.vendor.snowflake.lifecycle.SnowflakeVisitorFactory;
+import io.openlineage.spark.agent.vendor.snowflake.lifecycle.plan.SnowflakeEventHandlerFactory;
+import io.openlineage.spark.api.OpenLineageEventHandlerFactory;
+import io.openlineage.spark.api.Vendor;
+import java.util.Optional;
+
+public class SnowflakeVendor implements Vendor {
+
+  public static boolean hasSnowflakeClasses() {
+    /**
+     * Checking the Snowflake class with both
+     * SnowflakeRelationVisitor.class.getClassLoader.loadClass and
+     * Thread.currentThread().getContextClassLoader().loadClass. The first checks if the class is
+     * present on the classpath, and the second one is a catchall which captures if the class has
+     * been installed. This is relevant for Azure Databricks where jars can be installed and
```

Comment:
Is Azure Databricks relevant here, or just a doublecheck for safety sake?

Reply:
This part was already there I didn't touch it https://github.com/OpenLineage/OpenLineage/blob/99fed92fe2a8c63f24accbd8b632b15b72cce7c0/integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/SnowflakeRelationVisitor.java#L64